### ISSUE TITLE
About context.pop()

### ISSUE THREAD

**Issue by 1003581** [jkjung-avt/tensorrt_demos#148]
Hello,I create gpu context as follow,

Idevice = cuda.Device(0)
context = device.make_context()

//trt inferencing

context.pop()
del context

but " context.pop() " dose not work，return “”PyCUDA ERROR: The context stack was not empty upon module cleanup. “.  
If I ues  context.detach()，it work. I have not idea about this situation. Do you konw why?

**Comment by jkjung-avt** [comment]
Maybe you could read [PyCUDA documentation](https://documen.tician.de/pycuda/driver.html#devices-and-contexts).

**Comment by jkjung-avt** [comment]
> but " context.pop() " does not work，return “PyCUDA ERROR: The context stack was not empty upon module cleanup. “.

Please check whether you have done additional context.push()'s in the code.  I guess you need to do 1 pop() for each push() you've called.
